pages/studentPortal/quizcomponent/LangQuizRoute.tsx

```diff
@@ -11,91 +11,177 @@ export enum Stage {
+      { name: "Mobile development", result: "JavaScript", weight: 1 },
+      { name: "Data analytics", result: "Python", weight: 1 },
+      { name: "Product management", result: "JavaScript", weight: 1 },
+      { name: "Digital marketing", result: "JavaScript", weight: 1 },
+      { name: "UX/UI design", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
     ],
     [
-      "Websites",
-      "Storefront",
-      "Mobile apps",
-      "Games",
-      "Email",
-      "Tools to automate your life",
-      "I don't know / Not sure yet",
+      { name: "Websites", result: "JavaScript", weight: 1 },
+      { name: "Storefront", result: "HTML/CSS", weight: 1 },
+      { name: "Mobile apps", result: "JavaScript", weight: 1 },
+      { name: "Games", result: "JavaScript", weight: 1 },
+      { name: "Email", result: "HTML/CSS", weight: 1 },
+      { name: "Tools to automate your life", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
     ],
   ];
-  // create results state object that
-  // create custom type -- based on schema type in database
+
+  type QuizContext = {
+    title: string;
+    body: string;
+  };
+
+  const context: QuizContext[] = [
+    {
+      title: "What Coding Language Should I Learn First?",
+      body: "Take this free quiz to find out what coding language you should learn first.",
+    },
+    {
+      title: "Why do you want to learn coding?",
+      body: "Select all that apply.",
+    },
+    {
+      title: "What field of work are you interested in?",
+      body: "Select 1-3 choices.",
+    },
+    {
+      title: "What are you interested in building?",
+      body: "Select all that apply.",
+    },
+  ];
+
+  const selectFieldName = quiz.map((outerItem) => {
+    return outerItem.map((innerItem) => {
+      return innerItem.name;
+    });
+  });
+
+  const [stageResponses, setStageResponses] = useState([]);
+
+  const [score, setScore] = useState({
+    JavaScript: 0,
+    HTMLCSS: 0,
+    Python: 0,
+  });
+
+  const finalScore = (userSelections: string | any[]) => {
+    for (let i = 0; i < userSelections.length; i++) {
+      if (userSelections[i] && userSelections[i].result === "JavaScript") {
+        setScore((score) => ({
+          ...score,
+          JavaScript: score.JavaScript + userSelections[i].weight,
+        }));
+      }
+      if (userSelections[i] && userSelections[i].result === "HTMLCSS") {
+        setScore((score) => ({
+          ...score,
+          HTMLCSS: score.HTMLCSS + userSelections[i].weight,
+        }));
+      }
+      if (userSelections[i] && userSelections[i].result === "Python") {
+        setScore((score) => ({
+          ...score,
+          Python: score.Python + userSelections[i].weight,
+        }));
+      }
+    }
+  };
+  const userSelections: QuizSelection[] = [];
+  const productResult = () => {
```

Comment:
can you rename this to something more descriptive?

Reply:
Went with filteredQuizSelectionArr!